### ISSUE TITLE
Rule proposal: no-one-time-vars

### ISSUE THREAD

**Issue by Bas950** [sindresorhus/eslint-plugin-unicorn#1599]
### Description

Errors when a variable is only used once.

### Fail

```js
var once = 'used';
console.log(once);
```

### Pass

```js
var once = 'used';
console.log(once);
// Do other stuff
console.log(once);
```

**Comment by yakov116** [comment]
I don't think this is a good idea many times a var can be used hold data so when run in a function it's not called more then once.

**Comment by sindresorhus** [comment]
A variable declaration is also often used for readability. In many cases, variables with good names reduce the need for code comments.

**Comment by Bas950** [comment]
![image](https://user-images.githubusercontent.com/34792838/141342624-9318f9c3-8da6-4b3a-93da-ee94dfb15819.png)
This is a perfect example of why I suggested it.
So many useless consts...

**Comment by fisker** [comment]
I found this example is good..

Can you show us the version you want.

**Comment by sindresorhus** [comment]
The code in the screenshot has a lot of problems, but I don't think using some variable declarations is one of them.

**Comment by Bas950** [comment]
> I found this example is good..
> 
> Can you show us the version you want.

the first `if` can just be changed to
```
return document
    .querySelector('body > div > div > div > section > div > section > h1')
    ?.textContent?.match(/\s[A-Za-záàâãéèêíïóôõöúçÁÀÂÃÉÈÍÏÓÔÕÖÚÇ-]{3,}/)
    ?.shift()
  ```

**Comment by Bas950** [comment]
> The code in the screenshot has a lot of problems, but I don't think using some variable declarations is one of them.

Yea it's a PR I got on one of my repos, changes have been requested. But still in the first `if` case they are a lot of "useless" variables.

**Comment by sindresorhus** [comment]
We are going to pass on this. While I agree that sometimes code may have too many intermediate variables, there is simply no way to reliably detect when it's moot and when it's helpful for readability.
